pkg/river/internal/value/value_test.go

```diff
@@ -13,29 +13,85 @@ import (
 // implement an interface for a pointer receiver are retained correctly
 // throughout values.
 func TestInterfacePointerReceiver(t *testing.T) {
-	t.Run("Encode", func(t *testing.T) {
-		pm := &pointerMarshaler{}
-
-		val := value.Encode(pm)
-		require.Equal(t, value.TypeString, val.Type())
-		require.Equal(t, "Hello, world!", val.Text())
-	})
-
-	t.Run("From field", func(t *testing.T) {
-		type Body struct {
-			Data pointerMarshaler `river:"data,attr"`
-		}
-
-		b := &Body{}
+	tt := []struct {
+		name         string
+		encodeTarget any
+		key          string
+
+		expectBodyType value.Type
+		expectBodyText string
+
+		expectKeyExists bool
+		expectKeyValue  value.Value
+		expectKeyType   value.Type
+	}{
+		{
+			name:           "Encode",
+			encodeTarget:   &pointerMarshaler{},
+			expectBodyType: value.TypeString,
+			expectBodyText: "Hello, world!",
+		},
```

Comment:
Can we split this out into its own test function? It feels a bit weird that the rest of these tests are more about testing key lookups but this one isn't